app/subscribers/base_subscriber.rb

```diff
@@ -0,0 +1,11 @@
+class BaseSubscriber
+  def self.call(*args)
+    event = ActiveSupport::Notifications::Event.new(*args)
+    Rails.logger.info("#{self.name}: #{event.name} event received")
+    new.call(event)
+  end
+
+  def call(event)
+    raise NotImplementedError
+  end
+end
```

Comment:
Nitpick: I wonder if we should move this to Foreman namespace, chances are Rails might introduce this name in the future.